[ci-visibility] Separate excluded branches and tracked files into different sections (#20195)
DataDog/documentation: content/en/continuous_integration/intelligent_test_runner/_index.md

Intelligent Test Runner works by analyzing your test suite to determine the code each test covers, and then cross-referencing that coverage with the files impacted by a new code change. Datadog uses this information to run a selection of relevant, impacted tests, omitting the ones unaffected by the code change and reducing the overall testing duration.

By minimizing the number of tests run per commit, Intelligent Test Runner reduces the frequency of [flaky tests][1] disrupting your pipelines. This can be particularly frustrating when the test flaking is unrelated to the code change being tested. After enabling Intelligent Test Runner for your test services, you can limit each commit to its relevant tests to ensure that flaky tests unrelated to your code change don't end up arbitrarily breaking your build.

### Limitations

There are known limitations in the current implementation of Intelligent Test Runner that can cause it to skip tests that should be run under certain conditions. Specifically, Intelligent Test Runner is not able to detect changes in library dependencies, compiler options, external services or changes to data files in data-driven tests.

> To override Intelligent Test Runner and run all tests, add `ITR:NoSkip` (case insensitive) anywhere in your Git commit message. It is also possible to add a list of [excluded branches](#excluded-branches) and [tracked files](#tracked-files).

Reviewer: ```suggestion
To override Intelligent Test Runner and run all tests, add `ITR:NoSkip` (case insensitive) anywhere in your Git commit message. You can also add a list of [excluded branches](#excluded-branches) and [tracked files](#tracked-files).
```

## Set up a Datadog library

> Prior to setting up Intelligent Test Runner, you must set up [Test Visibility][2] for your particular language. If you are reporting data through the Agent, use v6.40 or 7.40 and later.

Reviewer: ```suggestion
Before setting up Intelligent Test Runner, you must configure [Test Visibility][2] for your particular language. If you are reporting data through the Agent, use v6.40 or 7.40 and later.
```
